recipes-multimedia/imx-dsp/imx-dsp_1.2.0.bb

```diff
@@ -0,0 +1,57 @@
+# Copyright 2017-2021 NXP
+
+DESCRIPTION = "i.MX DSP Wrapper, Firmware Binary, Codec Libraries"
+LICENSE = "Proprietary"
+LIC_FILES_CHKSUM = "file://COPYING;md5=03bcadc8dc0a788f66ca9e2b89f56c6f"
+
+inherit fsl-eula-unpack autotools pkgconfig
+
+SRC_URI = "${FSL_MIRROR}/${BP}.bin;fsl-eula=true"
+
+SRC_URI[md5sum] = "238d3f0256573ca657228d7090bcb7d3"
+SRC_URI[sha256sum] = "13f67f267d6d33e8be2a6c258a46cde3667258ac86435776cbf1a370de611476"
+
+EXTRA_OECONF = "-datadir=${base_libdir}/firmware --bindir=/unit_tests ${@bb.utils.contains('TUNE_FEATURES', 'aarch64', '--enable-armv8', ' ', d)}"
+
+RDEPENDS:${PN} += " imx-dsp-codec-ext"
+
+FILES:${PN} = "${libdir}/imx-mm/audio-codec/dsp \
+               ${libdir}/imx-mm/audio-codec/wrap \
+               ${base_libdir}/firmware/imx/dsp \
+               /unit_tests \
+"
+
+HIFI4_BIN ?= "hifi4_imx8qmqxp.bin"
+HIFI4_BIN:mx8qm = "hifi4_imx8qmqxp.bin"
+HIFI4_BIN:mx8qxp = "hifi4_imx8qmqxp.bin"
+HIFI4_BIN:mx8mp = "hifi4_imx8mp.bin"
+HIFI4_BIN:mx8ulp = "hifi4_imx8ulp.bin"
+
+# No need to do install about fsl_unia.h & fsl_types.h, which are duplicate with the ones' in imx-codec
+do_install:append () {
+    if [ -d ${D}/usr/include/imx-mm/audio-codec ]; then
+        rm -rf ${D}/usr/include/imx-mm/audio-codec
+    fi
```

Comment:
Please move the comment close to the `rm` command; also, please avoid checking if it exists and use `rm -r` only so if it stops of being installed we will know.

Reply:
@otavio I copied this recipe from [meta-imx](https://source.codeaurora.org/external/imx/meta-imx/tree/meta-bsp/recipes-multimedia/imx-dsp/imx-dsp_1.2.0.bb?h=hardknott-5.10.72-2.2.0)

As of the latest revision, `${D}/usr/include/imx-mm/audio-codec` is not being installed:
```
# After disabling do_install:append and rebuilding:
$ find tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/usr
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/usr/lib
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/usr/lib/imx-mm
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/usr/lib/imx-mm/audio-codec
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/usr/lib/imx-mm/audio-codec/wrap
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/usr/lib/imx-mm/audio-codec/wrap/lib_dsp_wrap_arm_elinux.so.1
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/usr/lib/imx-mm/audio-codec/wrap/lib_dsp_wrap_arm_elinux.so
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/usr/lib/pkgconfig
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/usr/lib/pkgconfig/libfsldspframework.pc
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/lib
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/lib/firmware
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/lib/firmware/imx
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/lib/firmware/imx/dsp
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/lib/firmware/imx/dsp/hifi4_imx8qmqxp.bin
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/lib/firmware/imx/dsp/hifi4_imx8mp.bin
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/lib/firmware/imx/dsp/hifi4_imx8ulp.bin
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/unit_tests
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/unit_tests/DSP
tmp/work/imx8mp_lpddr4_evk-fslc-linux/imx-dsp/1.2.0-r0/image/unit_tests/DSP/dsp_test.out
```
It is not clear if this is left over from an earlier revision, or in preparation for a future revision. Do you prefer we keep the logic, or remove it for now? Changing to `rm -r ${D}/usr/include/imx-mm/audio-codec` will error right now.

Reply:
Please remove; this is why we should review.